opt-may-2016/anusha/com/anusha/training/OptJuneTest.java

```diff
@@ -0,0 +1,22 @@
+package com.anusha.training;
+
+import java.util.Iterator;
+import java.util.TreeSet;
+
+public class OptJuneTest {
+
+	public static void main(String[] args) {
+		TreeSet set = new TreeSet(new LastNameComparator());    

```

Comment:
Always try to use referential polymorphism to initialize any collection. So it should be ideally `Set set = new TreeSet(new LastNameComparator());  `